src/app/records/search-records.component.ts

```diff
@@ -0,0 +1,119 @@
+import { ChangeDetectionStrategy, Component, Injector, OnInit } from '@angular/core';
+import { RecordDataForSearch, RecordQueryFilters, RecordResult, CustomFieldDetailed, RecordLayoutEnum } from 'app/api/models';
+import { RecordsService } from 'app/api/services';
+import { BaseSearchPageComponent } from 'app/shared/base-search-page.component';
+import { Observable } from 'rxjs';
+import { HttpResponse } from '@angular/common/http';
+import { HeadingAction } from 'app/shared/action';
+import { ApiHelper } from 'app/shared/api-helper';
+import { RecordHelperService } from 'app/core/records-helper.service';
+
+type RecordSearchParams = RecordQueryFilters & { owner: string, type: string, keywords: string };
+
+@Component({
+  selector: 'search-records',
+  templateUrl: 'search-records.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class SearchRecordsComponent
+  extends BaseSearchPageComponent<RecordDataForSearch, RecordSearchParams, RecordResult>
+  implements OnInit {
+
+  type: string;
+  param: string;
+  fieldsInSearch: Array<CustomFieldDetailed>;
+  fieldsInList: Array<CustomFieldDetailed>;
+
+  constructor(
+    injector: Injector,
+    private recordsService: RecordsService,
+    private recordsHelper: RecordHelperService
+  ) {
+    super(injector);
+  }
+
+  ngOnInit() {
+    super.ngOnInit();
+
+    this.param = this.route.snapshot.params.owner;
+    this.type = this.route.snapshot.params.type;
+
+    // Get search data
+    this.addSub(this.recordsService.getRecordDataForOwnerSearch({ owner: this.param, type: this.type }).subscribe(data => {
+      if (data.type.layout !== RecordLayoutEnum.LIST) {
+        throw new Error(`Invalid record layout: ${data.type.layout}`);
+      }
+
+      // TODO handle profileFields
+      this.fieldsInSearch = data.customFields.filter(cf => data.fieldsInSearch.includes(cf.internalName));
+      this.fieldsInList = data.customFields.filter(cf => data.fieldsInList.includes(cf.internalName));
+      this.form.setControl('customValues', this.fieldHelper.customValuesFormGroup(this.fieldsInSearch, {
+        useDefaults: false
+      }));
+      // Patch value to avoid the form reload twice
+      this.form.patchValue(data.query, { emitEvent: false });
```

Comment:
Nowadays it is no longer needed to use `emitEvent: false` because the changes are only observed after this method completes.
Actually, the `form.setControl` above would already generate a new request if this wasn't the case.